tests/app/authentication/test_authentication.py

```diff
@@ -372,3 +372,121 @@ def test_allow_valid_ips_bits(restrict_ip_sms_app):
     )
 
     assert response.status_code == 200
+
+
+@pytest.fixture
+def route_secret_app_with_key_1_only():
```

Comment:
A test with neither key1 or 2?

Reply:
Modified auth.py to return error. Added a new fixture and test.